src/main/java/com/kaleido/cabinet/client/CabinetClientProperties.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019. Kaleido Biosciences. All Rights Reserved
+ */
+
+package com.kaleido.cabinet.client;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.retry.annotation.EnableRetry;
+
+/**
+ * Holds properties from externalized configuration such as environment variables or {@code application.properties} files.
+ * Any variable beginning with {@code cabinet.client} or for environment variables {@code CABINET_CLIENT} and matching
+ * a field of this class will over-ride the default value. For example, {@code cabinet.client.password=secret} will set
+ * the value of the password used by the client.
+ * Default values are set to interact with the {@code dev} profile of Cabinet and are suitable for development and testing
+ * purposes.
+ */
+@ConfigurationProperties("cabinet.client")
+@EnableRetry
+public class CabinetClientProperties {
+
+    /*
+     * The properties username, password and base assume a localhost server where those credentials would be expected
```

Comment:
I think this configuration information would be better served by being in the README.md file at the root of this project under a section called configuration or something similar, so that we can have in a central place all environment variables that need to be set or overridden.